master/pkg/tasks/task_trial.go

```diff
@@ -100,15 +100,18 @@ func (s TrialSpec) ToTaskSpec(keys *ssh.PrivateAndPublicKeys) TaskSpec {
 	res.Entrypoint = []string{"/run/determined/train/entrypoint.sh"}
 
 	envVars := map[string]string{
-		"DET_EXPERIMENT_ID":      strconv.Itoa(s.ExperimentID),
-		"DET_TRIAL_ID":           strconv.Itoa(s.TrialID),
-		"DET_TRIAL_RUN_ID":       strconv.Itoa(s.TrialRunID),
-		"DET_TRIAL_SEED":         strconv.FormatUint(uint64(s.TrialSeed), 10),
-		"DET_EXPERIMENT_CONFIG":  jsonify(s.ExperimentConfig),
-		"DET_HPARAMS":            jsonify(s.HParams),
-		"DET_STEPS_COMPLETED":    strconv.Itoa(s.StepsCompleted),
-		"DET_UNIQUE_PORT_OFFSET": strconv.Itoa(trialUniquePortOffset(s.Base.Devices)),
-		"DET_TASK_TYPE":          string(model.TaskTypeTrial),
+		"DET_EXPERIMENT_ID":     strconv.Itoa(s.ExperimentID),
+		"DET_TRIAL_ID":          strconv.Itoa(s.TrialID),
+		"DET_TRIAL_RUN_ID":      strconv.Itoa(s.TrialRunID),
+		"DET_TRIAL_SEED":        strconv.FormatUint(uint64(s.TrialSeed), 10),
+		"DET_EXPERIMENT_CONFIG": jsonify(s.ExperimentConfig),
+		"DET_HPARAMS":           jsonify(s.HParams),
+		"DET_STEPS_COMPLETED":   strconv.Itoa(s.StepsCompleted),
+		"DET_TASK_TYPE":         string(model.TaskTypeTrial),
+		// DET_UNIQUE_PORT_OFFSET will be overwritten by the agent resource manager when the
+		// container is started, but only on agent resource pools.  This default value will apply
```

Comment:
It is actually overrides by slurm too, just in a startup hook we run for slurm stuff.

Reply:
Not sure I fully understand.  What should this comment say then?

Reply:
Was just a note that this is not the value slurm uses.